src/boundary_condition.cpp

```diff
@@ -37,7 +37,7 @@ void VacuumBC::handle_particle(Particle& p, const Surface& surf) const
 
 void ReflectiveBC::handle_particle(Particle& p, const Surface& surf) const
 {
-  Direction u = surf.reflect(p.r(), p.u(), &p);
+  Direction u = surf.reflect(p.r(), p.u());
```

Comment:
Just caught this. I think the last parameter still needs to be here for DAGMC surfaces.